tools/specialcasings.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+#-------------------------------------------------------------------------------------------------------
+# Copyright (C) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+#-------------------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+import json
+import os
+import shutil
+import urllib2
+from argparse import ArgumentParser
+
+def to_unicode_escape(codepoints):
+    return "".join(map(lambda codepoint: "\\u" + codepoint, codepoints.split(" ")))
```

Comment:
Why do we need to do this manually? Does the json module not escape correctly?

Reply:
To clarify: could we build up an actual Unicode string using `unichr` and do escaping upon output?

Reply:
Sorry for the terse and unhelpful comments, I was on my phone. I was just brainstorming ways we wouldn't have to do the step at the end where we change double-backslash to single-backslash because that feels mildly dangerous: we have valid JSON and then put it through a text filter that _could_ in some cases produce invalid JSON. I know that the _current_ content is fine with that transformation, but maybe we could build something that is more robust against future modification. I thought we could do something here like `"".join(map(lambda codepoint: unichr(int(codepoint)), codepoints.split(" ")))`, which would build up an actual unicode Python string instead of one containing backslash characters. Then during the output step we just trust `json.dumps` to put backslash characters in the right places. The current version obviously works fine, so feel free to leave as-is if you prefer it.

---
In reply to: [183217085](https://github.com/Microsoft/ChakraCore/pull/5026#discussion_r183217085) [](ancestors = 183217085)